contracts/OffsetHelper.sol

```diff
@@ -311,6 +311,11 @@ contract OffsetHelper is OffsetHelperStorage {
             block.timestamp
         );
 
+        // remove remaining approval if less input token was consumed
+        if (amounts[0] < amountIn) {
+            IERC20(_fromToken).approve(sushiRouterAddress, 0);
```

Comment:
why not re-set the approval regardless tho? 🤔

Reply:
Because it saves gas. Cheaper to do a memory variables comparison first. But I can remove the check if you prefer simpler logic.

Reply:
This was also actually just a quick fix ;) It would be even better to use [`SafeERC20.approve`](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#SafeERC20) here and/or set a max allowance once all together (this should then be done once when initializing eligible swappaple tokens).

Reply:
Yeah, let's keep the modification you've done so far and note all of these as follow-ups. The PR is already lengthy/complex enough & I don't feel like we need to dive deeper into this now.